app/main.py

```diff
@@ -1,3 +1,14 @@
 def cache(func):
-    # Write your code here
-    pass
+    history = {}
+
+    def wrapper(*args, **kwargs):
+        nonlocal history
```

Comment:
You don't need to use nonlocal here. It can be used for reassigning variables. If you need just to change some fields of dict it is unnecessary.